packages/plugins/typescript/urql-graphcache/src/index.ts

```diff
@@ -0,0 +1,244 @@
+  if (isWrappingType(type)) {
+    return unwrapType(type.ofType);
+  }
+
+  return type || null;
+};
+
+function constructType(
+  typeNode: TypeNode,
+  schema: GraphQLSchema,
+  nullable = true,
+  allowString = false
+): string {
+  switch (typeNode.kind) {
+    case 'ListType': {
+      return nullable
+        ? `Maybe<Array<${constructType(typeNode.type, schema, false, allowString)}>>`
+        : `Array<${constructType(typeNode.type, schema, false, allowString)}>`;
+    }
+    case 'NamedType': {
+      const type = schema.getType(typeNode.name.value);
+      if (!type.astNode) {
+        return nullable ? `Maybe<Scalars['${typeNode.name.value}']}>` : `Scalars['${typeNode.name.value}']`;
+      }
+
+      switch (type.astNode.kind) {
+        case 'EnumTypeDefinition':
+        case 'UnionTypeDefinition':
+        case 'InputObjectTypeDefinition':
+        case 'ObjectTypeDefinition':
+        case 'ScalarTypeDefinition': {
+          const finalType = `RequireFields<${typeNode.name.value}, '__typename'>${allowString ? ' | string' : ''}`;
+          return nullable ? `Maybe<${finalType}>` : finalType;
+        }
+        case 'InterfaceTypeDefinition': {
+          const possibleTypes = schema.getPossibleTypes(type as GraphQLAbstractType).map(function contruct(x) {
+            return `RequireFields<${x}, '__typename'>`;
+          });
+          const finalType = allowString ? possibleTypes.join(' | ') + ' | string' : possibleTypes.join(' | ');
+          return nullable ? `Maybe<${finalType}>` : finalType;
+        }
+      }
+      break;
+    }
+    case 'NonNullType': {
+      return constructType(typeNode.type, schema, false, allowString);
+    }
+  }
+}
+
+const capitalize = (s: string): string => s.charAt(0).toUpperCase() + s.slice(1);
+
+function getKeysConfig(schema: GraphQLSchema) {
+  const keys = [];
+
+  const roots = [
+    schema.getQueryType()?.name,
+    schema.getMutationType()?.name,
+    schema.getSubscriptionType()?.name,
+  ].filter(Boolean);
+
+  const typemap = schema.getTypeMap();
+  const isValidType = (type: string) => !roots.includes(type) && !type.startsWith('__') && !baseTypes.includes(type);
+
+  Object.keys(typemap).forEach(function (key) {
+    if (!isValidType(key)) return;
+
+    const type = typemap[key];
+    if (type.astNode?.kind !== Kind.OBJECT_TYPE_DEFINITION) return;
+
+    keys.push(`${type.name}?: (data: RequireFields<${type.name}, '__typename'>) => null | string`);
+  });
+
+  return `
+type GraphCacheKeysConfig = {
+  ${keys.join('\n  ')}
+}
+  `;
+}
+
+function getResolversConfig(schema: GraphQLSchema) {
+  const resolvers = [];
+
+  const roots = [schema.getMutationType()?.name, schema.getSubscriptionType()?.name].filter(Boolean);
+
+  const typemap = schema.getTypeMap();
+  const isValidType = (type: string) => !roots.includes(type) && !type.startsWith('__') && !baseTypes.includes(type);
+
+  Object.keys(typemap).forEach(function (key) {
+    if (!isValidType(key)) return;
```

Comment:
This should probably change to detecting whether it's not a scalar or internal type 😅